app/controllers/reports_controller.rb

```diff
@@ -143,7 +143,7 @@ def report_params
       :unreviewed, :filter, :question_labels, :show_question_labels, :question_order, :text_responses,
       :percent_type, :unique_rows, :calculations, :option_set, :mission_id, :mission,
       :disagg_question_id, :group_by_tag,
-      option_set_choices_attributes: %i[option_set_id],
-      calculations_attributes: %i[id type report_report_id attrib1_name question1_id arg1 attrib1 question1 rank _destroy])
+      option_set_choices_attributes: %i[id option_set_id _destroy],
+      calculations_attributes: %i[id type attrib1_name question1_id arg1 attrib1 question1 rank _destroy])
```

Comment:
I don't quite understand why this fixes the bug?

Reply:
Yeah it's a bit esoteric. With Rails nested params, if you don't provide an ID to match on when you're updating, it creates a new record. Since we weren't permitting the ID to come through, Rails was happily creating a new join table record on each update instead of updating the existing one.

In newer versions of Rails, it complains loudly if it rejects a param. But in earlier versions it fails silently (probably due to the fact that strong parameters was a major major breaking change when it first came out so they were introducing it slowly).